packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -501,8 +501,7 @@ actual object RealmInterop {
                     }
                 }
                 is RealmObjectInterop -> {
-                    val nativePointer = (value as RealmObjectInterop).`$realm$ObjectPointer`
-                        ?: error("Cannot add unmanaged object")
+                    val nativePointer = (value as RealmObjectInterop).objectPointer
```

Comment:
Isn't the cast automatic inside the `is RealmObjectInterop` ?

Reply:
Yes, it is done automatically by the when. I didn't noticed that it could be removed.